### PR TITLE
test: add test for issue 1468

### PR DESCRIPTION
**Pull request by JoanFM**
**Changes introduced**
Test showing problem with #1468 

It hangs unless wither of the 2 things happen:
- Remote _merge_chunks
- Reduce parallel to 1

### CODE REVIEW

File: tests/integration/issues/github_1468/flow.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    uses: pods/filter_chunk_1.yml
+    read_only: true
+  filter_chunk_2:
+    uses: pods/filter_chunk_1.yml
+    read_only: true
+    needs: [gateway]
+  merge_route_process:
+    uses: _forward
+    parallel: 2
+    uses_before: _merge_chunks
```
Comment (hanxiao) [inline tests/integration/issues/github_1468/flow.yml:16]: is it intentional, you could write `uses: _merge_chunks`

is it suggesting the problem only exists when you set both?

Comment (JoanFM) [reply tests/integration/issues/github_1468/flow.yml:16]: `uses` could be anything else. 
I want `_merge_chunks` to accumulate all the `chunks` from the 2 incoming paths in the HeadPea, but at the same time HeadPea must handle Routing.

Comment (hanxiao) [reply tests/integration/issues/github_1468/flow.yml:16]: but why a `needs` here does not fit your need?

```yaml
  merge_route_process:
    uses: _merge_chunks
    parallel: 2
#    uses_before: _merge_chunks
    needs: [filter_chunk_1, filter_chunk_2]

```

Comment (JoanFM) [reply tests/integration/issues/github_1468/flow.yml:16]: ah will this merge everything? then yes I will test tomorrow. But then the load balancing could fail to send every pair of chunks to the same shard right?

Comment (hanxiao) [reply tests/integration/issues/github_1468/flow.yml:16]: no i mean you can always separate it into two pods, e.g.
```yaml
!Flow
version: '1.0'
pods:
  - name: filter_chunk_1
    uses: pods/filter_chunk_1.yml
    read_only: true
  - name: filter_chunk_2
    uses: pods/filter_chunk_1.yml
    read_only: true
    needs: [gateway]
  - needs: [filter_chunk_1, filter_chunk_2]
  - name: merge_route_process
    parallel: 2

```

Comment (JoanFM) [reply tests/integration/issues/github_1468/flow.yml:16]: ok yes good. However there is big way for improvement. The big issue that I see is that then just changing the amount of parallelization to the same Pod is not enough which adds complexity to the user

**Comment by JoanFM** [comment]
I still think @hanxiao it can make sense to merge it and look at it and see if we can improve this

**Review by hanxiao — APPROVED** [review]

**Pull request merged**